_cli/data_platform_workflows_cli/craft_tools/promote_legacy.py

```diff
@@ -0,0 +1,314 @@
+        )
+    commit_sha = commit_shas.pop()
+    logging.info(f"All revisions on {repr(channel)} were built from git commit {repr(commit_sha)}")
+    return commit_sha, revisions
+
+
+@dataclasses.dataclass(kw_only=True)
+class Metadata:
+    name: str
+    display_name: str
+    oci_resources: dict[str, str]
+
+    @classmethod
+    def from_file(cls, *, directory: pathlib.Path):
+        file = yaml.safe_load((directory / "metadata.yaml").read_text())
+        # (Only for Kubernetes charms) get OCI resources
+        oci_resources = {}
+        for resource_name, resource in file.get("resources", {}).items():
+            if resource["type"] != "oci-image":
+                continue
+            oci_hash = resource["upstream-source"]
+            if "@sha256:" not in oci_hash:
+                raise ValueError(
+                    "Unable to promote charm that does not pin all of its `oci-image` resources "
+                    f"to a sha256 digest in metadata.yaml: {repr(resource['upstream-source'])}"
+                )
+            oci_resources[resource_name] = oci_hash
+        return cls(
+            name=file["name"], display_name=file["display-name"], oci_resources=oci_resources
+        )
+
+
+def charm():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--track", required=True)
+    parser.add_argument("--from-risk", required=True)
+    parser.add_argument("--to-risk", required=True)
+    parser.add_argument("--ref", required=True)
+    args = parser.parse_args()
+    directory = pathlib.Path(".")
+
+    track = args.track
+    if track == "":
+        raise ValueError("`track` input must not be empty string")
+    if "/" in track:
+        raise ValueError(f"`track` input cannot contain '/' character: {repr(track)}")
+    from_risk = Risk.get(args.from_risk, direction=Direction.FROM)
+    to_risk = Risk.get(args.to_risk, direction=Direction.TO)
+    if not to_risk < from_risk:
+        raise ValueError(
+            f"`to-risk` input ({repr(to_risk.value)}) must be lower risk than `from-risk` input "
+            f"({repr(from_risk.value)})"
+        )
+    if to_risk is Risk.STABLE and from_risk is not Risk.CANDIDATE:
+        raise ValueError(
+            "Only the 'candidate' risk can be promoted to 'stable'. Promote "
+            f"{repr(from_risk.value)} to 'candidate' first"
+        )
+
+    if not args.ref.startswith("refs/heads/"):
+        raise ValueError(
+            "This workflow must be run on `workflow_dispatch` from the branch that contains track "
+            f"{repr(track)}"
+        )
+
+    if not pathlib.Path(".github/release.yaml").exists():
+        raise FileNotFoundError(
+            "Repository must contain `.github/release.yaml` to automatically generate release "
+            "notes in the correct format. See "
+            "https://github.com/canonical/data-platform-workflows/blob/main/.github/workflows/promote_charm.md#step-3-add-githubreleaseyaml-file"
```

Comment:
issue: this link is not found now, + suggest to link to the `_promote_charm_legacy.md` instead